src/components/Search.tsx

```diff
@@ -14,38 +14,31 @@ type SearchProps = {
   location: string;
   setLocation: (place: Geoname) => void;
 };
+
 export const Search = ({location, setLocation}: SearchProps) => {
   const geonames = useMemo(() => getGeonames(), []);
-  const map = geonames.reduce(
-    (acc: Record<string, Geoname>, val) => ((acc[val.name] = val), acc),
-    {},
-  );
-
-  const dataset = geonames.map((val, idx) => ({
-    id: idx.toString(),
-    title: val.name,
-  }));
+  const dataset = useMemo(() => getDataset(geonames), [])
 
   const handleSelect = (item: TAutocompleteDropdownItem) => {
     if (item && item.title !== null) {
-      setLocation(map[item.title]);
+      setLocation(geonames[item.title]);
     }
   };
 
   return (
     <GlassView containerStyle={styles.container} glassStyle={styles.glassCcontainer} blurStyle={{blurAmount: 25, blurType: 'light'}}>
       <AutocompleteDropdown
         clearOnFocus={false}
-        closeOnBlur={true}
+        closeOnBlur={false}
         closeOnSubmit={false}
         textInputProps={{ placeholder: 'Search location', placeholderTextColor: 'white', style: styles.textStyle }}
         onSelectItem={handleSelect}
-        dataSet={[{id: '9_000_000_000', title: location}, ...dataset]}
+        dataSet={[...dataset]}
         inputContainerStyle={styles.searchBar}
-        debounce={3000}
+        debounce={1500}
         showChevron={false}
         showClear={false}
-        RightIconComponent={<TouchableOpacity onPress={() => {}}><Icon source={locationAnchor} size={24}/></TouchableOpacity>}
+        // RightIconComponent={<TouchableOpacity onPress={() => {}}><Icon source={locationAnchor} size={24}/></TouchableOpacity>}
```

Comment:
This one is commented out?

I guess it should still be there, and a bug report filled to make it work?

Reply:
Already an issue for it. Forgot to add a link to the issue though. Done now.

Reply:
Ah. I misunderstood what you said. I think its better to remove it when its not working, and add it if its important enough to fix now. If its not important enough, we will have a nonfunctional button there .